### PR TITLE
Align multimodal module with evaluation logic

### PR DESCRIPTION
**Pull request by suzhoum**
Issue #, if available:

Description of changes:

This PR introduced `framework` and `constraint` concepts to running `multimodal` module, so that the output result captures more structured metadata information and aligns with AMLB benchmark output format. In this way, we can reuse the existing evaluation logic. It supports both custom framework and custom constraint.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sample_configs/tabular_cloud_configs.yaml
Comment (yinweisu) [inline sample_configs/tabular_cloud_configs.yaml:1]: Looks like cloud config has been c hanged. Do we need to update the script to generate config too?

File: src/autogluon/bench/frameworks/multimodal/exec.py [hunk 1]
```diff
@@ -135,9 +131,8 @@ def run(
                             multimodal_dataset_registry.list_keys()
 
         benchmark_dir (str): The path to the directory where benchmarking artifacts should be saved.
-        time_limit (int): The maximum amount of time (in seconds) to spend training the predictor (default: 10).
-        presets (str): The name of the AutoGluon preset to use (default: "None").
-        hyperparameters (dict): A JSON of hyperparameters to use for training (default: None).
+        constraint (str): The resource constraint used by benchmarking during AWS mode.
```
Comment (yinweisu) [inline src/autogluon/bench/frameworks/multimodal/exec.py:134]: Maybe specify what would happen if no constraint is being passed in? i.e. what's the default constraint

**Review by yinweisu — APPROVED** [review]
LGTM! Couple minor comments

File: sample_configs/tabular_cloud_configs.yaml
Comment (suzhoum) [reply sample_configs/tabular_cloud_configs.yaml:1]: Yes I'll update the generation script in the following PR.

**Pull request merged**